src/__tests__/BrahmosES6class.test.js

```diff
@@ -0,0 +1,354 @@
+      }
+      render() {
+        return (
+          <Inner name={this.state.bar} onClick={this.handleClick.bind(this)} />
+        );
+      }
+    }
+    test(<Foo initialValue="foo" />, 'DIV', 'foo');
+    attachedListenerWithCallback(()=>expect(renderedName).toBe('bar'));
+    // Passed the test as a callback
+  });
+
+  it('should not implicitly bind event handlers', () => {
+    class Foo extends Brahmos.Component {
+      constructor(props) {
+        super(props);
+        this.state = {bar: props.initialValue};
+      }
+      handleClick() {
+        this.setState({bar: 'bar'});
+      }
+      render() {
+        return <Inner name={this.state.bar} onClick={this.handleClick} />;
+      }
+    }
+    test(<Foo initialValue="foo" />, 'DIV', 'foo');
+    expect(attachedListener).toThrow();
+  });
+
+  it('will call all the normal life cycle methods', () => {
+    let lifeCycles = [];
+    class Foo extends Brahmos.Component {
+      constructor() {
+        super();
+        this.state = {};
+      }
+      // UNSAFE_componentWillMount() {
+      //   lifeCycles.push('will-mount');
+      // }
+      componentDidMount() {
+        lifeCycles.push('did-mount');
+      }
+      // UNSAFE_componentWillReceiveProps(nextProps) {
+      //   lifeCycles.push('receive-props', nextProps);
+      // }
+      shouldComponentUpdate(nextProps, nextState) {
+        lifeCycles.push('should-update', nextProps, nextState);
+        return true;
+      }
+      // UNSAFE_componentWillUpdate(nextProps, nextState) {
+      //   lifeCycles.push('will-update', nextProps, nextState);
+      // }
+      componentDidUpdate(prevProps, prevState) {
+        lifeCycles.push('did-update', prevProps, prevState);
+      }
+      componentWillUnmount() {
+        lifeCycles.push('will-unmount');
+      }
+      render() {
+        return <span className={this.props.value} />;
+      }
+    }
+    class Outer extends Brahmos.Component{
+      constructor(props){
+        super(props);
+        this.state={
+          isFooVisible: this.props.visible
+        }
+      }
+      unmountFoo(callback){
+        this.setState({
+          isFooVisible: false
+        },()=>callback)
+      }
+      render(){
+        if(this.state.isFooVisible){
+          return <Foo value={this.props.value}/>;
+        }
+        return <div/>;
+      }
+    }
+    test(<Outer visible value="foo" />, 'SPAN', 'foo');
+    expect(lifeCycles).toEqual(['did-mount']);
+    lifeCycles = []; // reset
+    const instance = test(<Outer visible value="bar" />, 'SPAN', 'bar');
+    expect(lifeCycles).toEqual([
+      'should-update', freeze({value: 'bar'}), {},
+      'did-update', freeze({value: 'foo'}), {},
+    ]);
+    lifeCycles = []; // reset
+    instance.unmountFoo(()=>expect(lifeCycles).toEqual(['will-unmount']));
+  });
+
+});
+
+
+/** Currently unsupported tests */
+
+// Brahmos hasn't installed prop-types
+
+// it('renders based on context in the constructor', () => {
```

Comment:
We can remove this commented test.
brahmos will not support this, in favour of new contextAPI. The legacy context pattern is buggy.